DOCS-5144 Adds Build an Integration, Build a Marketplace Offering, and Create a Tile Documentation (#14297)
DataDog/integrations-core: docs/dev/README.md

* Reading the documentation about creating [Agent-based integrations][2].

For more information about sell a Datadog integration, see [Build a Marketplace Offering][4].

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: https://docs.datadoghq.com/developers/integrations/api_integration/
[2]: https://docs.datadoghq.com/developers/integrations/agent_integration/
[3]: https://docs.datadoghq.com/integrations/
[4]: https://docs.datadoghq.com/developers/integrations/marketplace_offering/
[5]: https://partners.datadoghq.com/
[6]: https://docs.datadoghq.com/account_management/users/#add-new-members-and-manage-invites
[7]: https://learn.datadoghq.com/courses/intro-to-integrations
[8]: https://learn.datadoghq.com/
[9]: https://docs.datadoghq.com/developers/authorization/
[10]: https://app.datadoghq.com/integrations
[11]: https://docs.datadoghq.com/agent/
[12]: https://docs.datadoghq.com/api/latest/
[13]: https://docs.datadoghq.com/developers/authorization/
[14]: https://docs.datadoghq.com/metrics/custom_metrics/
[15]: https://docs.datadoghq.com/getting_started/site/

> [16]: https://docs.datadoghq.com/help/

Author: ```suggestion
[16]: https://docs.datadoghq.com/help/
[17]: https://app.datadoghq.com/marketplace
```

> [16]: https://docs.datadoghq.com/help/

Author: ```suggestion
[16]: https://docs.datadoghq.com/help/
[17]: https://app.datadoghq.com/marketplace
```

---
title: Build an Integration
kind: documentation
description: Learn how to develop and publish an offering on the Integrations page.
aliases:
- /guides/agent_checks/
- /agent/agent_checks
- /developers/agent_checks/

> dependencies: "https://github.com/DataDog/integrations-core/blob/alai97/add-marketplace-documentation/docs/dev/README.md"

Author: ```suggestion
dependencies: "https://github.com/DataDog/integrations-core/blob/master/docs/dev/README.md"
```

> further_reading:

Reviewer: Can we add the "Create an Agent Integration" and "Create an API Integration" docs to the "Further Reading" section?

> - link: "/developers/integrations/marketplace_offering/"

Author: ```suggestion
- link: "/developers/integrations/agent_integration/"
   tag: "Documentation"
   text: "Create an Agent integration"
- link: "/developers/integrations/api_integration/"
   tag: "Documentation"
   text: "Create an API integration"
- link: "/developers/integrations/marketplace_offering/"
```

### API-based integrations

API-based integrations can submit telemetry—such as metrics, traces, logs, and more—from external platforms using the [Datadog API][12]. Customers can then visualize and correlate this data alongside data from the rest of their stack, allowing them to quickly analyze and remediate issues. API-based integrations may also read data out of Datadog once customers [authorize access using OAuth][13]. 

Technology Partners write and host the implementation code that makes up the integration. [Creating an API integration][1] work well for Technology Partners building a connector between Datadog and another SaaS platform.

### Benefits

By creating an integration, you can achieve the following benefits:

Correlate your data with user observability data
: Leverage Datadog to increase the value of your platform by allowing customers to see the data from your platform alongside the rest of their technology stack.

> Increase mean-time-to-resolution (MTTR) for customers 

Author: ```suggestion
Decrease mean-time-to-resolution (MTTR) for customers 
```

: When a customer's account is enriched with data from an integration, they are able to see a broader view of their entire stack, allowing them to debug and remediate issues more quickly. 

Increase adoption and visibility 
: Ensuring native functionality for Datadog reduces friction to adoption, and displaying a tile on the [Integrations page][10] or the [Marketplace page][17] provides key visibility to all of Datadog's customers.

> Submit out-of-the-box metrics at no extra cost 

Reviewer 2: Opted to remove this as it's misleading for marketplace listings

---
title: Build an Integration
kind: documentation
description: Learn how to develop and publish an offering on the Integrations page.
aliases:
- /guides/agent_checks/
- /agent/agent_checks
- /developers/agent_checks/
dependencies: "https://github.com/DataDog/integrations-core/blob/alai97/add-marketplace-documentation/docs/dev/README.md"
further_reading:
- link: "/developers/"
  tag: "Documentation"
  text: "Learn how to develop on the Datadog platform"
---

## Overview

This page walks through how Technology Partners can build an out-of-the-box integration using the Datadog Agent or Datadog API. 

The [Integrations page][3] includes integrations and custom dashboard widgets built by both Datadog and our Technology Partners, available at no cost to Datadog customers. 

> The [Datadog Marketplace][4], on the other hand, is a commercial platform for Technology Partners to _sell_ a variety of offerings, including integrations, custom dashboard widgets, software subscriptions/licenses, and professional services to Datadog customers.

Reviewer 3: Decided on license

## Datadog Integrations

### [API-based Integration][1]
API-based integrations can enrich a customer's Datadog environment by submitting data from external technologies via the Datadog API. Customers authorize API integrations access to their accounts through OAuth, and partners write and host the implementation code that makes up the integration. API integrations work well for partners building a connector between Datadog and another SaaS platform.

> ### [Agent-based Integration][2]

Reviewer: Can we switch the ordering and have the "Agent-based Integration" section go before "API-based Integration," since in the initial paragraph we mention Agent Integrations first?

> Agent-based integrations use the Datadog Agent to submit data via checks written by the partner. The implementation code for these integrations is hosted by Datadog. Agent integrations are best suited for collecting data from custom systems or applications. Writing an agent integration requires you to publish and deploy your solution as a Python wheel (.whl).

Reviewer: Can we update this to: 

"Agent Integrations are best suited for collecting data from systems or applications that live in a local area network (LAN) or virtual private cloud (VPC)." 

Separately, can we keep consistency in capitalizing the "A" in Agent.

## Why create an integration?

> Out-of-the-box Metrics - Metrics reported from official Datadog integrations (unless the integration is sending in potentially unlimited metrics) are not counted as custom metrics, and therefore won't impact a customer's custom metric allocation.

Reviewer: Should we bold the first reason for readability? E.g., "**Adoption** - Ensuring..."

I also think Discoverability should go first. 

These reasons could be fleshed out a bit more. For example, I don't see anything about the main benefits of observability, correlation, resolving issue for users, etc.

Reviewer 2: My latest commit didn't sync up to staging yet, but i bolded those terms :) 

@emarsha94 for the comment on fleshing out the reasons

Adoption - Ensuring native support for Datadog reduces friction to adoption and incentivizes Datadog customers to build out their technology stack with partner technologies.

Added Visibility - Partners integrations appear on the Integrations page alongside all Datadog-built integrations, providing key visibility to all of Datadog's customers.

### Responsibilities

> Going forward, you, as the author of the integration, are the active maintainer of the integration. You are responsible for maintaining the code and ensuring the integration’s functionality. There is no specific time commitment, but you must be a maintainer of the code for the foreseeable future. Datadog extends support on a best-effort basis for partner-built integrations, so please reach out to Datadog support if help is needed.

Reviewer: We can probably delete the following line to pare things down. I don't think it adds much information:

"There is no specific time commitment, but you must be a maintainer of the code for the foreseeable future."

## Let's Get Started 

> To create an API integration, [click here][1].

Reviewer: Let's keep the ordering consistent throughout the page. I.e., either API integrations are mentioned first each time, or Agent integrations are.

> To create an Agent-based integration, [click here][2].

Reviewer 4: Are these on new lines? It's on the same line in the preview but it would be easier to read if it's on a new line.

{{< partial name="whats-next/whats-next.html" >}}

> [1]: API Based link
> [2]: Agent based link 
> [3]: Integrations page link
> [4]: Marketplace link

Author: ```suggestion
[1]: https://docs.datadoghq.com/developers/integrations/api_integration/
[2]: https://docs.datadoghq.com/developers/integrations/agent_integration/
[3]: https://docs.datadoghq.com/integrations/
[4]: https://docs.datadoghq.com/developers/integrations/marketplace_offering/
```

Whereas the **Marketplace** page is a commercial platform for Datadog customers and Technology Partners to buy and sell a variety of offerings including Agent-based or API-based integrations, Datadog Apps, software subscriptions/licenses, and professional services.

## Datadog Integrations

[API-based Integration][1]
This is to enrich and submit data from your backend. API integrations work well in building a connector between Datadog and another SaaS platform. Partners will write and host their code within their enviornment.

[Agent-based Integration][2]
A full data integration that includes code and uses the Datadog Agent to collect data. 

## Why create an integration?

Out-of-the-box Metrics - Metrics reported from accepted integrations are not counted as custom metrics, and therefore don’t impact your custom metric allocation. (Integrations that emit potentially unlimited metrics may still be considered custom.) 

Adoption - Ensuring native support for Datadog reduces friction to adoption, and incentivizes people to use your product, service, or project. 

> Added Visibility - Partners are featured within the Datadog ecosystem. 

Reviewer 3: Partners will be featured on the itnegrations page adding visibility

{{< whatsnext desc="See the following integration documentation to get started:" >}}
  {{< nextlink href="/developers/integrations/create_a_tile" >}}Create an Integration Tile {{< /nextlink >}}
  {{< nextlink href="/developers/integrations/api_integration" >}}Create an Agent Integration {{< /nextlink >}}

> {{< /whatsnext >}}

Reviewer 3: This was removed and replaced by below

> ### Responsibilities

Reviewer 3: I removed the previous header and put this one instead

further_reading:
- link: "/developers/"
  tag: "Documentation"
  text: "Learn how to develop on the Datadog platform"
---

## Overview

This page walks through how Technology Partners can build an out-of-the-box integration using the Datadog Agent or Datadog API. 

The [Integrations page][3] includes integrations built by both Datadog and our Technology Partners, available at no cost to Datadog customers. 

The [Datadog Marketplace][4], on the other hand, is a commercial platform for Technology Partners to _sell_ a variety of offerings, including integrations, software licenses, and professional services to Datadog customers.

## Datadog Integrations

### [Agent-based Integration][2]
Agent-based integrations use the Datadog Agent to submit data via checks written by the partner. The implementation code for these integrations is hosted by Datadog. Agent Integrations are best suited for collecting data from systems or applications that live in a local area network (LAN) or virtual private cloud (VPC). Writing an Agent integration requires you to publish and deploy your solution as a Python wheel (.whl).

### [API-based Integration][1]

> API-based integrations can enrich a customer's Datadog environment by submitting data from external technologies via the Datadog API. Customers authorize API integrations access to their accounts through OAuth, and partners write and host the implementation code that makes up the integration. API integrations work well for partners building a connector between Datadog and another SaaS platform.

Reviewer: Can we replace the hazy "enrich a customer's Datadog environment" and be more specific about submitting telemetry like metrics, logs, traces that can be correlated, alerted on, visualized, and analyzed? 

Separately, are we avoiding stating the option to also read data out, not just submit?

## Why create an integration?

**Correlate your data with user observability data** - Leverage Datadog to increase the value of your platform by  allowing customers to see the data from your platform alongside the rest of their technology stack.

> **Increase MTTR for customers** - When a customer's account is enriched with data from an integration, they are able to see a broader view of their entire stack, allowing them to debug and remediate issues more quickly. 

Reviewer 4: Does everyone know what MTTR stands for?

> **Increase Adoption and Visibility** - Ensuring native support for Datadog reduces friction to adoption, and displaying a tile on our Integrations page provides key visibility to all of Datadog's customers.

Reviewer: I think "Adoption" and "Visibility" should be lowercase

> **Submit out-of-the-box metrics at no extra cost** - Metrics reported from official Datadog integrations (unless the integration is sending in potentially unlimited metrics) are not counted as custom metrics, and therefore won't impact a customer's custom metric allocation.

Reviewer 4: Should we also call out billing?

### Responsibilities

> Going forward, you, as the author of the integration, are the active maintainer of the integration. You are responsible for maintaining the code and ensuring the integration’s functionality. Datadog extends support on a best-effort basis for partner-built integrations, so please reach out to Datadog support if help is needed.

Reviewer: Here, we write "Datadog extends **support on a best-effort basis** for partner-built integration" but just a few lines above we say a benefit is "**Ensuring native support for Datadog reduces friction to adoption**..."

> Going forward, you, as the author of the integration, are the active maintainer of the integration. You are responsible for maintaining the code and ensuring the integration’s functionality. Datadog extends support on a best-effort basis for partner-built integrations, so please reach out to Datadog support if help is needed.

Reviewer: ```suggestion
As the author of the integration, you are responsible for maintaining the code and ensuring the integration functions properly across Datadog's regions. Please reach out to Datadog's Support team if help is needed.
```

> ## Join the Datadog partner network

Reviewer: Can we reorder to:

Overview > Join the Datadog Partner Network > Why create an integration > Datadog integrations > Let's get started?

Could we also keep headings consistent with capitalizations? For example, "Datadog Partner Network" should be capitalized but only the "Let's" in "Let's Get Started" should be capitalized.

Before listing an integration on Datadog, you will first need to apply to the [Datadog Partner Network's][3] **Technology Partner** track. Once your application has been approved, you can begin to develop your integration.

## Request a sandbox account

All Technology Partners can request a dedicated sandbox Datadog account to aid in their development. This sandbox account has a free license that you can use to send in data, build out dashboards, and more. 

To request a sandbox account:

1. Log into the [Datadog Partner Portal][6].
2. On your personal homepage, click on the **Learn More** button under **Sandbox Access**.
3. Select **Request Sandbox Upgrade**.

<div class="alert alert-info">If you are already a member of a Datadog organization (including a trial org), you may need to switch to your newly created sandbox. For more information, see the <a href="https://docs.datadoghq.com/account_management/org_switching/">Account Management documentation</a>.</div>

> Creating a developer sandbox may take up to one or two business days. Once your sandbox is created, you can [invite new members from your organization][7] to collaborate with.

Reviewer: This link isn't working correctly in the preview

## Let's Get Started 
To create an Agent-based integration, [click here][2].

> To create an API integration, [click here][1].

Reviewer 4: Might be worth calling out the Marketplace here if we intend for this to be the landing page for partners because we also offer software license listings and professional services.

further_reading:
- link: "/developers/integrations/marketplace_offering/"
  tag: "Documentation"
  text: "Learn how to sell an integration on the Datadog Marketplace"
- link: "/developers/"
  tag: "Documentation"
  text: "Learn how to develop on the Datadog platform"
---

## Overview

This page walks you through how Technology Partners can [build an out-of-the-box integration](#create-a-datadog-integration) using the [Datadog Agent][11] or the [Datadog API][12], and list their offering on the **Integrations** or **Marketplace** page. 

{{< tabs >}}
{{% tab "Integrations" %}}

The [Integrations page][101] includes integrations built by both Datadog and our Technology Partners, available at _no cost_ to Datadog customers. 

{{< img src="developers/integrations/integrations_overview.png" alt="The Datadog Integrations page" style="width:100%;" >}}

[101]: https://app.datadoghq.com/integrations

{{% /tab %}}
{{% tab "Marketplace" %}}

The [Marketplace page][101] is a commercial platform for Technology Partners to _sell_ a variety of offerings, including integrations, software licenses, and professional services to Datadog customers.

> {{< img src="developers/marketplace/marketplace_overview.png" alt="The Datadog Marketplace page" style="width:100%" >}}

Author: ```suggestion
{{< img src="developers/marketplace/marketplace_updated_overview.png" alt="The Datadog Marketplace page" style="width:100%" >}}
```
